### ISSUE TITLE
False Positive: NonSolvableDep

### ISSUE THREAD

**Issue by ezzieyguywuf** [pkgcore/pkgcheck#281]
Steps to reproduce
============

```sh
git clone https://github.com/gentoo-haskell/gentoo-haskell.git
cd gentoo-haskell/dev-lang/purescript
pkgcheck scan
```

Expected Behaviour
============

Since `dev-lang/purescript` is in `profiles/profile.mask`, I expect pkgcheck to recognize and downgrade this from an error to a warning.

Repoman does something like this - `repoman full -dx` does **not** produce an error, but does provide some output that suggests running with `--without-mask`.

Consquently, `repoman full -dx --without-mask` **does** produce the error.

Actual Behaviour
===========

pkgcheck produces an error instead of a warning.

**Comment by radhermit** [comment]
I'll also note you might find it more useful to base CI configs for https://github.com/gentoo-haskell/gentoo-haskell/pull/1149 off the official example in the https://github.com/pkgcore/pkgcheck-ci repo at https://github.com/pkgcore/pkgcheck-ci/blob/main/.github/workflows/pkgcheck.yml rather than whatever overlay you probably copied that from.

Feel free to submit PRs to the pkgcheck-ci repo if you think the example is missing anything that's generically useful.

**Comment by ezzieyguywuf** [comment]
Well noted, thank you Tim!

**Comment by ezzieyguywuf** [comment]
Are you suggesting that I copy/paste the yaml file you linked?

Is there anything else in the linked repo that i might need?

**Comment by radhermit** [comment]
> Are you suggesting that I copy/paste the yaml file you linked?

Yes, I was suggesting you start with that as a base instead of whatever you started with as it's the canonical example config for github that enables things like git historical repo checks, metadata caching, internalized syncing, colorized output, etc and avoids the various repo hacks using mkdir/wget seen in your PR.

I imagine the only thing that requires customization would be adding check/keyword filtering and any other options as wanted to the related ``pkgcheck scan`` line.

> Is there anything else in the linked repo that i might need?

No, that repo is just a basic test for various CI service config examples using pkgcheck. Just grab the one related to the service you want and off you go. It could of course use some more docs to relay that to potential users, but I think that's already an open issue. :smile:
